lib/matplotlib/image.py

```diff
@@ -486,16 +486,30 @@ def _make_image(self, A, in_bbox, out_bbox, clip_bbox, magnification=1.0,
                     raise ValueError("Invalid dimensions, got %s" % (A.shape,))
 
                 output = np.zeros((out_height, out_width, 4), dtype=A.dtype)
+                output_a = np.zeros((out_height, out_width, 4), dtype=A.dtype)
 
                 alpha = self.get_alpha()
                 if alpha is None:
-                    alpha = 1.0
+                    alpha = 1
+
+                #resample alpha channel
+                alpha_tuple = (A[...,3], A[...,3], A[...,3])
+                alpha_channel = _rgb_to_rgba(np.dstack(alpha_tuple))
+                _image.resample(
```

Comment:
This will also re sample 2D (1-channel) arrays so we can do this without making a a 4 channel copy.

Reply:
This is still doing the alpha re-sampling 4 times instead of once.

Reply:
sorry, was reading through commit-by commit and thought I was looking at the total diff, ignore my (now deleted) comment if you got an email.